plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/dialogs/ColorsAnnotationHelper.java

```diff
@@ -0,0 +1,220 @@
+     * It does not change the session model.
+     * 
+     * @param colorsAnnotationSourceName
+     *            the source annotation used to retrieve the associated {@link DAnnotationEntry}.
+     * @return the {@link DAnnotationEntry} associated to the specified color category.
+     */
+    private DAnnotationEntry getOrCreateColorAnnotationEntry(String colorsAnnotationSourceName) {
+        DAnnotationEntry dAnnotationEntry = null;
+        if (this.optDanalysis.isPresent()) {
+            dAnnotationEntry = getColorAnnotationEntry(colorsAnnotationSourceName).orElseGet(() -> {
+                DAnnotationEntry annotationEntry = DescriptionFactory.eINSTANCE.createDAnnotationEntry();
+                annotationEntry.setSource(colorsAnnotationSourceName);
+                return annotationEntry;
+            });
+        }
+        return dAnnotationEntry;
+    }
+
+    /**
+     * Gets the {@link DAnnotationEntry} for the specified source annotation.
+     * 
+     * @param colorsAnnotationSourceName
+     *            the source annotation used to retrieve the associated {@link DAnnotationEntry}.
+     * @return the optional {@link DAnnotationEntry} associated to the specified color category.
+     */
+    private Optional<DAnnotationEntry> getColorAnnotationEntry(final String colorsAnnotationSourceName) {
+        if (this.optDanalysis.isPresent()) {
+            final List<DAnnotationEntry> sessionAnnotations = optDanalysis.get().getEAnnotations();
+            for (final DAnnotationEntry annotationEntry : sessionAnnotations) {
+                if (annotationEntry.getSource().equals(colorsAnnotationSourceName)) {
+                    return Optional.of(annotationEntry);
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Sets the details of the {@link DAnnotationEntry} with the specified colorsList.
+     * 
+     * @param colorsAnnotationSourceName
+     *            the source annotation used to retrieve the associated {@link DAnnotationEntry}.
+     * @param colorsList
+     *            the list of colors to set for the specified color category.
+     */
+    public void setColorsDetails(String colorsAnnotationSourceName, List<RGB> colorsList) {
+        session.getTransactionalEditingDomain().getCommandStack().execute(new RecordingCommand(session.getTransactionalEditingDomain()) {
+            @Override
+            protected void doExecute() {
+                DAnnotationEntry colorsEntry = getOrCreateColorAnnotationEntry(colorsAnnotationSourceName);
+                List<String> stringColorsList = convertRGBToString(colorsList);
+                List<String> colorDetails = colorsEntry.getDetails();
+                colorDetails.clear();
+                colorDetails.addAll(stringColorsList);
+
+                if (colorDetails.isEmpty()) {
+                    removeColorAnnotation(colorsEntry);
```

Comment:
no because it is a good thing to have this annotation. In collab, It limits the case of object locked by another.
At the contrary, we could even have added a migratipn participant(but do not do it, it is over kill)